ghost-cat-app/src/app/View/search-page/search-page.component.html

```diff
@@ -31,8 +31,10 @@ <h3>Standard Filters</h3>
     <mat-form-field *ngIf="searchByCamera" class="marginLeft">
       <mat-label>Camera Trap</mat-label>
       <mat-select [(value)]="cameraTrap" [required]="true">
```

Comment:
I think you can make this a multiselect list by just changing this to be:
`<mat-select ... multiple>`

Check out the documentation for `mat-select` here: https://material.angular.io/components/select/overview#multiple-selection

I think using this built in version would probably fix up the user experience bugs, and it might make it easier to get the map selection working as well :)